File: scripts/test_db.py
#!/usr/bin/env python3
import os
from pathlib import Path
from dotenv import load_dotenv
from supabase import create_client

# Load environment variables from .env.local
env_path = Path(__file__).parent.parent / '.env.local'
load_dotenv(env_path)

supabase_url = os.getenv('NEXT_PUBLIC_SUPABASE_URL')
supabase_key = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

print(f"Supabase URL: {'Set' if supabase_url else 'NOT SET'}")
print(f"Service Role Key: {'Set' if supabase_key else 'NOT SET'}")

if not supabase_url or not supabase_key:
    print("Error: Missing environment variables")
    exit(1)

supabase = create_client(supabase_url, supabase_key)

# Get conversations
print("\n=== CONVERSATIONS ===")
result = supabase.table('conversations').select('*').order('updated_at', desc=True).limit(10).execute()
conversations = result.data
print(f"Found {len(conversations)} conversations")
for c in conversations:
    print(f"- {c['id']}: \"{c['title']}\" ({c.get('message_count', 0)} msgs)")

# Check for specific conversation
target_conv_id = '100831e3-16a8-4df8-9f7e-5b17193ac9fb'
print(f"\n=== TARGET CONVERSATION ({target_conv_id}) ===")
result = supabase.table('conversations').select('*').eq('id', target_conv_id).execute()
if result.data:
    print(f"Found: {result.data[0]}")
else:
    print("NOT FOUND")

# Get messages for target conversation
print("\n=== MESSAGES IN TARGET CONVERSATION ===")
result = supabase.table('messages').select('id, role, content, created_at').eq('conversation_id', target_conv_id).order('created_at').execute()
messages = result.data
print(f"Found {len(messages)} messages")
for m in messages:
    preview = (m['content'] or '[no content]')[:80]
    print(f"- [{m['role']}] {preview}...")

# Get ALL messages count
print("\n=== ALL MESSAGES COUNT ===")
result = supabase.table('messages').select('*', count='exact').limit(5).execute()
print(f"Total messages in database: {result.count}")
print(f"Sample messages: {len(result.data)}")
if result.data:
    print("\nSample message data:")
    for m in result.data:
        print(f"  Conv: {m['conversation_id']}, Role: {m['role']}")

File: scripts/test_db.ts
import { createClient } from '@supabase/supabase-js'
import * as dotenv from 'dotenv'
import { resolve } from 'path'

// Load environment variables from .env.local
dotenv.config({ path: resolve(__dirname, '../.env.local') })

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY!

console.log('Supabase URL:', supabaseUrl ? 'Set' : 'NOT SET')
console.log('Service Role Key:', supabaseKey ? 'Set' : 'NOT SET')

async function testDb() {
  const supabase = createClient(supabaseUrl, supabaseKey)

  // Get conversations
  const { data: conversations, error: convError } = await supabase
    .from('conversations')
    .select('*')
    .order('updated_at', { ascending: false })
    .limit(10)

  console.log('\n=== CONVERSATIONS ===')
  if (convError) {
    console.error('Error fetching conversations:', convError)
  } else {
    console.log('Found', conversations?.length, 'conversations')
    conversations?.forEach(c => {
      console.log(`- ${c.id}: "${c.title}" (${c.message_count} msgs)`)
    })
  }

  // Check for specific conversation
  const targetConvId = '100831e3-16a8-4df8-9f7e-5b17193ac9fb'
  const { data: targetConv, error: targetError } = await supabase
    .from('conversations')
    .select('*')
    .eq('id', targetConvId)
    .single()

  console.log('\n=== TARGET CONVERSATION ===')
  if (targetError) {
    console.error('Error fetching target conversation:', targetError)
  } else {
    console.log('Conversation:', targetConv)
  }

  // Get messages for target conversation
  const { data: messages, error: msgError } = await supabase
    .from('messages')
    .select('id, role, content, created_at')
    .eq('conversation_id', targetConvId)
    .order('created_at', { ascending: true })

  console.log('\n=== MESSAGES IN TARGET CONVERSATION ===')
  if (msgError) {
    console.error('Error fetching messages:', msgError)
  } else {
    console.log('Found', messages?.length, 'messages')
    messages?.forEach(m => {
      const preview = m.content ? m.content.substring(0, 80) : '[no content]'
      console.log(`- [${m.role}] ${preview}...`)
    })
  }

  // Get ALL messages to check if they exist
  const { data: allMessages, error: allMsgError, count } = await supabase
    .from('messages')
    .select('*', { count: 'exact' })
    .limit(5)

  console.log('\n=== ALL MESSAGES COUNT ===')
  if (allMsgError) {
    console.error('Error:', allMsgError)
  } else {
    console.log('Total messages in database:', count)
    console.log('Sample messages:', allMessages?.length)
    if (allMessages && allMessages.length > 0) {
      console.log('\nSample message data:')
      allMessages.forEach(m => {
        console.log(`  Conv: ${m.conversation_id}, Role: ${m.role}`)
      })
    }
  }
}

testDb().catch(console.error)
